lib/metadata/in_memory/backend.js

```diff
@@ -41,7 +42,8 @@ const metastore = {
             if (!metadata.buckets.has(bucketName)) {
                 return cb(errors.NoSuchBucket);
             }
-            return cb(null, metadata.buckets.get(bucketName));
+            return cb(null, BucketInfo.fromObj(
```

Comment:
@LaurenSpiegel the bucketinfo fromobj do nothing in the case of the bucketclient backend actually, because it take an BucketInfo already, and create a new BucketInfo from this one :)
@rahulreddy it should not be relate to old splitter compatibility, i will think about it but basically here, i just move the pparsing of BucketInfo in the memory backend :)